src/mediapipe_internal/mediapipegraphdefinition.hpp

```diff
@@ -93,19 +77,38 @@ class MediapipeGraphExecutor {
     const PipelineDefinitionStateCode getStateCode() const { return status.getStateCode(); }
     const model_version_t getVersion() const { return VERSION; }
     // METADATA
+private:
+    Status createInputsInfo();
+    Status createOutputsInfo();
+
+public:
     const tensor_map_t getInputsInfo() const;
     const tensor_map_t getOutputsInfo() const;
 
-    Status infer(const KFSRequest* request, KFSResponse* response, ExecutionContext executionContext, ServableMetricReporter*& reporterOut) const;
     // TODO simultaneous infer & reload handling
-    Status validate(ModelManager& manager);
+    Status create(std::shared_ptr<MediapipeGraphExecutor>& pipeline, const KFSRequest* request, KFSResponse* response);
 
-private:
-    Status createInputsInfo();
-    Status createOutputsInfo();
+    Status validate(ModelManager& manager);
```

Comment:
Lets organise those members and methods in order:
public static:
public:
protected:
private static:
private: